cmd/protoc-gen-go-grpc/grpc.go

```diff
@@ -121,6 +138,16 @@ func generateFile(gen *protogen.Plugin, file *protogen.File) *protogen.Generated
 		g.P("// source: ", file.Desc.Path())
 	}
 	g.P()
+	// Attach all comments associated with the package field
+	loc = file.Desc.SourceLocations().ByPath(protoreflect.SourcePath{fileDescriptorProtoPackageFieldNumber})
+	for _, s := range loc.LeadingDetachedComments {
+		g.P(protogen.Comments(s))
+		g.P()
+	}
+	if s := loc.LeadingComments; s != "" {
+		g.P(protogen.Comments(s))
+		g.P()
+	}
```

Comment:
Since this is now performed twice, please factor back out into a function.  (When it was only used once, it didn't make sense; I didn't realize it would ultimately be needed twice.) 

What does this second part handle, by the way?  We don't seem to have any examples in our repo that use it IIUC.

Reply:
The comments above each section show what it's grabbing. 
The first is for the comments preceding the `syntax` line. 
The second is for the comments preceding the `package` line.

(This behavior matches how protoc-gen-go works)

You can see the functionality by modifying the `reflection/grpc_testing/test.proto` file and running `regenerate.sh` to see the output.